haas/client/node.py

```diff
@@ -0,0 +1,99 @@
+import json
+from haas.client.base import ClientBase
+
+
+class Node(ClientBase):
+    """Consists of calls to query and manipulate node related
+
+    objects and relations.
+    """
+
+    def list(self, is_free):
+        """List all nodes that HIL manages """
+        url = self.object_url('nodes', is_free)
+        return self.check_response(self.httpClient.request('GET', url))
+
+    def show(self, node_name):
+        """Shows attributes of a given node """
+
+        url = self.object_url('node', node_name)
+        return self.check_response(self.httpClient.request('GET', url))
+
+    def register(self, node, subtype, *args):
+        """Register a node with appropriate OBM driver. """
+#       Registering a node requires apriori knowledge of the
+#       available OBM driver and its corresponding arguments.
+#       We assume that the HIL administrator is aware as to which
+#       Node requires which OBM, and knows arguments required
+#       for successful node registration.
+
+        obm_api = "http://schema.massopencloud.org/haas/v0/obm/"
+        obm_types = ["ipmi", "mock"]
+#       FIXME: In future obm_types should be dynamically fetched
+#        from haas.cfg, need a new api call for querying available
```

Comment:
Get rid of "from haas.cfg"; the API call makes more sense as you describe, but wherever we get it from it's not going to be the server's config file.